README.md

```diff
@@ -7,290 +7,505 @@
+    echo '<code>No `id_token` URL parameter!</code> ';
+    exit;
+}
+
+// Do we have a valid algorithm?
+if (empty($_GET[ 'token_alg' ]) || ! in_array($_GET[ 'token_alg' ], [ 'HS256', 'RS256' ])) {
+    echo '<code>Missing or invalid `token_alg` URL parameter!</code> ';
+    exit;
+}
 
-## Examples
+$config = [
+    'supported_algs' => [ $_GET[ 'token_alg' ] ],
+    'client_secret' => getenv('AUTH0_CLIENT_SECRET'),
+];
 
-Check the [basic-oauth](https://github.com/auth0/auth0-PHP/tree/master/examples/basic-oauth) example to see a quick demo on how the sdk works.
-You just need to create a `.env` file with the following information:
+if ('HS256' === $_GET[ 'token_alg' ]) {
+    $config['client_secret'] = getenv('AUTH0_CLIENT_SECRET');
+} else {
+    $config['authorized_iss'] = [ 'https://' . getenv('AUTH0_DOMAIN') . '/' ];
+}
 
+try {
+    $verifier = new JWTVerifier($config);
+    $decoded_token = $verifier->verifyAndDecode($_GET[ 'id_token' ]);
+    echo '<pre>' . print_r($decoded_token, true) . '</pre>';
+} catch (InvalidTokenException $e) {
+    echo 'Caught: InvalidTokenException - ' . $e->getMessage();
+} catch (CoreException $e) {
+    echo 'Caught: CoreException - ' . $e->getMessage();
+} catch (\Exception $e) {
+    echo 'Caught: Exception - ' . $e->getMessage();
+}
 ```
-AUTH0_CLIENT_SECRET=YOUR_APP_SECRET
-AUTH0_CLIENT_ID=YOU_APP_CLIENT
-AUTH0_DOMAIN=YOUR_DOMAIN.auth0.com
-AUTH0_CALLBACK_URL=http://localhost:3000/index.php
-AUTH0_APPTOKEN=A_VALID_APPTOKEN_WITH_CREATE_USER_SCOPE
+
+Additional parameters for the `JWTVerifier` configuration array are:
+
+- **cache**: Receives an instance of `Auth0\SDK\Helpers\Cache\CacheHandler` (Supported `FileSystemCacheHandler` and `NoCacheHandler`). Defaults to `NoCacheHandler` (RS256 only).
+- **guzzle_options**: Configuration propagated to Guzzle when fetching the JWKs (RS256 only). These options are [documented here](http://docs.guzzlephp.org/en/stable/request-options.html).
+- **secret\_base64\_encoded**: When `true`, it will decode the secret used to verify the token signature. This is only used for HS256 tokens and defaults to `true`. Your Application settings will say whether the Client Secret provided is encoded or not. 
+
+## Usage - Management API
+
+This SDK also provides a wrapper for the Management API, which is used to perform operations on your Auth0 tenant. Using this API, you can:
+
+- Search for and create users
+- Create and update Applications
+- Retrieve log entries
+- Manage rules 
+
+... and much more. See our [documentation](https://auth0.com/docs/api/management/v2) for information on what's possible and the examples below for how to authenticate and access this API. 
+
+### Authentication
+
+In order to use the Management API, you must authenticate one of two ways:
+
+- For temporary access or testing, you can [manually generate an API token](https://auth0.com/docs/api/management/v2/tokens#get-a-token-manually) and save it in your `.env` file
```

Comment:
I'd add a warning stating that the tokens cannot be revoked once generated, and that they will be valid until the expiration time specified on the text field next to the "regenerate token" button of the MGMT API's settings page. The doc page is already linked so that's OK.

Reply:
I'll leave the details to that docs link

Reply:
https://github.com/auth0/docs/pull/6138